File: aspnet-core/src/Timesheet.Application/APIs/Timesheets/Projects/Dto/ProjectDto.cs
using Abp.Application.Services.Dto;
using Abp.AutoMapper;
using Ncc.Entities;
using System;
using System.Collections.Generic;
using System.Text;
using static Ncc.Entities.Enum.StatusEnum;
using System.Linq;
using Ncc.IoC;

namespace Timesheet.Timesheets.Projects.Dto
{
    [AutoMapTo(typeof(Project))]
    public class ProjectDto : EntityDto<long>
    {
        public string Name { get; set; }
        public string Code { get; set; }
        public ProjectStatus Status { get; set; }       
        public DateTime TimeStart { get; set; }
        public DateTime? TimeEnd { get; set; }
        public string Note { get; set; }
        public ProjectType ProjectType { get; set; }
        public long CustomerId { get; set; }
        public List<ProjectTaskDto> Tasks { get; set; }
        public List<ProjectUsersDto> Users { get; set; }
        public List<ProjectTargetUserDto> ProjectTargetUsers { get; set; }
        public string KomuChannelId { get; set; }
        public bool IsNotifyToKomu { get; set; }
        public bool IsNoticeKMSubmitTS { get; set; } = false;
        public bool IsNoticeKMRequestOffDate { get; set; } = false;
        public bool IsNoticeKMApproveRequestOffDate { get; set; } = false;
        public bool IsNoticeKMRequestChangeWorkingTime { get; set; } = false;
        public bool IsNoticeKMApproveChangeWorkingTime { get; set; } = false;
        public bool isAllUserBelongTo { get; set; }
    }
}

File: angular/src/app/service/api/model/project-Dto.ts
export interface GetProjectDto {
    id: number;
    name: string;
    code: string;
    status: number;
    customerName: string;
}

export interface ProjectDto {
    id: number;
    name: string;
    code: string;
    projectType: number;
    status: number;
    note: string;
    timeStart: Date;
    timeEnd: Date;
    customerId: number;
    isAllUserBelongTo: boolean;
    tasks: TaskProjectDto[];
    users: UserProjectDto[];
    projectTargetUsers: ProjectTargetUsersDto[];
    komuChannelId: string;
    isNotifyToKomu: boolean;
    isNoticeKMSubmitTS: boolean;
    isNoticeKMRequestOffDate: boolean;
    isNoticeKMApproveRequestOffDate: boolean;
    isNoticeKMRequestChangeWorkingTime: boolean;
    isNoticeKMApproveChangeWorkingTime: boolean;
}
export interface CustomerProjectDto {
    customerName: string;
    projects: any[];
}
export interface TaskProjectDto {
    // projectId: number,
    taskId: number,
    billable: true,
    id?: number,
    name?: string
}
export interface UserProjectDto {
    userId: number,
    // projectId: number,
    type: number,
    id?: number
    isTemp: boolean
}
export interface ProjectTargetUsersDto {
    userId: number,
    roleName: string,
    id?: number
}

export class BaseUserDto {
    name: string;
    isActive: boolean;
    type: number;
    jobTitle: string;
    level: number;
    avatarPath: string;
    avatarFullPath: string;
    branch: number;
    emailAddress: string;
    branchId: number;
    branchDisplayName: string;
    branchColor: string;
}

export class UserDto extends BaseUserDto {
    id: number;
}

export class AddUserToRoleDTO {
    userId: number;
    role: string;
}

export class ProjectUserDto extends BaseUserDto {
    userId: number;
    ptype: number;
    isTemp: boolean;
}

export class DisplayProjectTargetUserDto extends BaseUserDto {
    userId: number;
    roleName: string;
}

export class ProjectTaskDto {
    taskId: number;
    name: string;
    billable: boolean;
    type: number;
}

export class QuantityProjectDto {
    status : number;
    quantity : number;
}
